python_modules/libraries/dagster-airflow/dagster_airflow/utils.py

```diff
@@ -0,0 +1,140 @@
+
+
+# Airflow DAG ids and Task ids allow a larger valid character set (alphanumeric characters,
+# dashes, dots and underscores) than Dagster's naming conventions (alphanumeric characters,
+# underscores), so Dagster will strip invalid characters and replace with '_'
+def normalized_name(name, unique_id=None):
+    base_name = "airflow_" + "".join(c if VALID_NAME_REGEX.match(c) else "_" for c in name)
+    if not unique_id:
+        return base_name
+    else:
+        return base_name + "_" + str(unique_id)
+
+
+@contextmanager
+def replace_airflow_logger_handlers():
+    prev_airflow_handlers = logging.getLogger("airflow.task").handlers
+    try:
+        # Redirect airflow handlers to stdout / compute logs
+        handler = logging.StreamHandler(sys.stdout)
+        handler.setFormatter(logging.Formatter(LOG_FORMAT))
+        root = logging.getLogger("airflow.task")
+        root.handlers = [handler]
+        yield
+    finally:
+        # Restore previous log handlers
+        logging.getLogger("airflow.task").handlers = prev_airflow_handlers
+
+
+def serialize_connections(connections):
+    serialized_connections = []
+    for c in connections:
+        serialized_connection = {
+            "conn_id": c.conn_id,
+            "conn_type": c.conn_type,
+        }
+        if hasattr(c, "login") and c.login:
+            serialized_connection["login"] = c.login
+        if hasattr(c, "password") and c.password:
+            serialized_connection["password"] = c.password
+        if hasattr(c, "host") and c.host:
+            serialized_connection["host"] = c.host
+        if hasattr(c, "schema") and c.schema:
+            serialized_connection["schema"] = c.schema
+        if hasattr(c, "port") and c.port:
+            serialized_connection["port"] = c.port
+        if hasattr(c, "extra") and c.extra:
+            serialized_connection["extra"] = c.extra
+        if hasattr(c, "description") and c.description:
+            serialized_connection["description"] = c.description
+        serialized_connections.append(serialized_connection)
+    return serialized_connections
+
+
+if os.name == "nt":
+    import msvcrt  # pylint: disable=import-error
+
+    def portable_lock(fp):
+        fp.seek(0)
+        msvcrt.locking(fp.fileno(), msvcrt.LK_LOCK, 1)
+
+    def portable_unlock(fp):
+        fp.seek(0)
+        msvcrt.locking(fp.fileno(), msvcrt.LK_UNLCK, 1)
+
+else:
+    import fcntl
+
+    def portable_lock(fp):
+        fcntl.flock(fp.fileno(), fcntl.LOCK_EX)
+
+    def portable_unlock(fp):
+        fcntl.flock(fp.fileno(), fcntl.LOCK_UN)
+
+
+class Locker:
+    def __init__(self, lock_file_path="."):
+        self.lock_file_path = lock_file_path
+        self.fp = None
+
+    def __enter__(self):
+        self.fp = open(f"{self.lock_file_path}/lockfile.lck", "w+", encoding="utf-8")
+        portable_lock(self.fp)
+
+    def __exit__(self, _type, value, tb):
+        portable_unlock(self.fp)
+        self.fp.close()
```

Comment:
Yeesh what is going on here. Did you vendor this from somewhere? I think this is worth checking in a separate file in its own PR that documents where it came from, has tests, etc.

Reply:
Oh this was already there hmmmm